Tests/Realm.Tests/Realm.Tests.csproj

```diff
@@ -140,6 +140,7 @@
     </EmbeddedResource>
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\..\Realm\Realm.SourceGenerator\Realm.SourceGenerator.csproj" />
```

Comment:
Should this be referencing the project like this? What about on CI - I imagine we want to do something similar to what we do with `Realm.csproj` where we pick it up from the nuget package.

Reply:
This file is heavily changed in the other PR, so that it uses the local reference locally and the nuget package on CI